server/src/test/java/io/druid/segment/realtime/plumber/RealtimePlumberSchoolTest.java

```diff
@@ -297,4 +319,144 @@ public void run()
 
     Assert.assertEquals(1, metrics.failedPersists());
   }
+
+  @Test(timeout = 60000)
+  public void testPersistHydrantGaps() throws Exception
+  {
+    final Object commitMetadata = "dummyCommitMetadata";
+    testPersistHydrantGapsHelper(commitMetadata);
+  }
+
+  private void testPersistHydrantGapsHelper(final Object commitMetadata) throws Exception
+  {
+    final MutableBoolean committed = new MutableBoolean(false);
+    Interval testInterval = new Interval(new DateTime("1970-01-01"), new DateTime("1970-12-31"));
```

Comment:
Intervals are exclusive of the end instant, so you probably want to end on new DateTime("1971-01-01")